src/controller.cpp

```diff
@@ -452,24 +451,29 @@ geometry_msgs::Twist Controller::update(
     current_tf.rotation.x, current_tf.rotation.y, current_tf.rotation.z, current_tf.rotation.w);
   current_with_carrot_.setRotation(cur_rot);
 
-  size_t path_pose_idx;
+  const auto current_tf_for_find_position = [&current_tf, track_base_link = config_.track_base_link,
```

Comment:
These lambda functions don't really increase readability. But I get the point.

Although `current_tf_for_find_position` sounds like something that could be a simple request function on the class?

Reply:
Actually, something is off here.

You're comment suggests a `current_tf` is requested. But there is also a variable `current_tf`. We should fix these names first.

Probably one tf points to `base_link` and the other to the tf that tries to follow the carrot. Right @cesar-lopez-mar ?

Reply:
Yes, if I move the lambda to a member function we could add some documentation (that was present in the original implementation).

`current_tf`` is used if `config_.track_base_link`, else the following is used:
```c++
    // find position of current position with projected carrot
    geometry_msgs::Transform current_with_carrot_g;
    tf2::convert(current_with_carrot_, current_with_carrot_g);
```

Reply:
> These lambda functions don't really increase readability. But I get the point.
> 
> Although `current_tf_for_find_position` sounds like something that could be a simple request function on the class?

I agree that readability does not improve for all developers that might no be familiar with lambda functions (including myself). I would keep the comments of the original implementation in this function to help understanding the new code.

Reply:
> Actually, something is off here.
> 
> You're comment suggests a `current_tf` is requested. But there is also a variable `current_tf`. We should fix these names first.
> 
> Probably one tf points to `base_link` and the other to the tf that tries to follow the carrot. Right @cesar-lopez-mar ?

Indeed current_tf points to base_link, and the other one to the carrot.

Reply:
Not the carrot. But the ControlPoint.

Reply:
I opened #87 for this. Referencing here to put it on the list ;)

Reply:
Okay, I'm confused as to what the conclusion of this discussion is/should be. I need some help.

Would replacing the lambda with a member function suffice? If so, I would like some help in naming and documenting that function.

Or, is there some underlying problem that needs to be fixed and which is preventing us from continuing with this PR?

Reply:
Yes, there is an underlying problem that the variable names are off. I propose we fix that first.

I hope a more clear solution direction will follow from that.

Reply:
These variable names are really confusing indeed. `current_tf` also seems to imply that this is a `tf2::Transform`, but looking at this code, that cannot be the case (otherwise this wouldn't compile, as `current_with_carrot_g` is a `geometry_msgs::Transform`).

BTW, I like this refactor with the IILE